modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -2384,6 +2388,8 @@ class TaskProcessor {
             collectOutputs(task)
         }
         catch ( Throwable error ) {
+            def trace = handler.getTraceRecord()
+            task.context.put('trace', trace)
```

Comment:
Not getting this. Why adding only in task context on there's an exception

Reply:
If I am not wrong, this is the case where the task is retried. In other cases it is going to finalize the task and I think there is no need to put the record in the context.

Reply:
I see your point. My point is that if this is going to be available only on error condition, we may want to expose using a different context e.g. `error` that could bring more information about the error condition + the `trace`

Reply:
Do you mean adding the another implicit variable called 'error' in the TaskContext and put as condition something like Throwable object apart form the 'trace'? Or include the 'trace' inside this 'error' variable?

Reply:
The only problem that I see of putting the property trace inside error is that it can be confused with the stack trace.  Maybe we could call the property 'traceRecord'

Reply:
In reality I think I would prefer something like `task.previous` to make it clear that it is information about the previous attempt that only exists when attempt > 1:

```groovy
task.previous.exitStatus
task.previous.errorMessage
task.previous.trace
```

This also keeps everything contained within the implicit variable `task`

Reply:
I like it. @pditommaso, what do you think?

Reply:
`previous` implies an ordering which is something it cannot be guaranteed. Maybe `task.last` or `task.failure` ?

Reply:
I prefer `task.last`.  As I mentioned before, associating `trace` with `task.failure` or `error` could be interpreted as the stack trace of the error

Reply:
thinking more, if this is going to be exposed on a task retry, I believe it would be better to use `task.failure` or a similar concept. `task.last` or `task.previous` makes me think it's available in any case

Reply:
Maybe we are over engineering here. Considering those variables are already in the `task` scope, does not make much sense adding a new `last` context including again the `exitStatus`. 

What about just adding `lastTrace` as it's done for `exitStatus`, `retryCount`, etc?

Reply:
That's fine too, but I would prefer `previousTrace` because `last` can mean either "previous" or "final", it often confuses people

Reply:
Can we also include the error as e.g. `task.previousError` ? Or at least the error message, to not expose runtime details. Since the original use case is to determine the type of error that occurred, it seems like the error message would be just as important as the trace

Reply:
Sure!